types/features/features.go

```diff
@@ -0,0 +1,102 @@
+// Package features provides the JSON structure that is printed in `runc features` (since runc v1.1.0)
+package features
+
+// Features is printed in `runc features`
+type Features struct {
+	// Capabilities is the list of the recognized capabilities , e.g., "CAP_SYS_ADMIN"
+	// Nil value means "unknown", not "no support for any capability".
+	Capabilities []string `json:"capabilities,omitempty"`
+
+	// MountOptions is the list of the recognized mount options, e.g., "ro"
+	// Nil value means "unknown", not "no support for any mount option".
+	MountOptions []string `json:"mountOptions,omitempty"`
+
+	Cgroup   Cgroup   `json:"cgroup"`
+	Seccomp  Seccomp  `json:"seccomp"`
+	Apparmor Apparmor `json:"apparmor"`
+	Selinux  Selinux  `json:"selinux"`
+	Criu     Criu     `json:"criu"`
+
+	// Annotations contains implementation-specific annotation strings,
+	// such as the implementation version, and third-party extensions.
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Seccomp represents the "seccomp" field
+type Seccomp struct {
+	// Enabled is true if seccomp support is compiled in
+	Enabled bool `json:"enabled"`
+
+	// Actions is the list of the recognized actions, e.g., "SCMP_ACT_NOTIFY"
+	// Nil value means "unknown", not "no support for any action".
+	Actions []string `json:"actions,omitempty"`
+
+	// Operators is the list of the recognized actions, e.g., "SCMP_CMP_NE"
+	// Nil value means "unknown", not "no support for any operator".
+	Operators []string `json:"operators,omitempty"`
+
+	// Operators is the list of the recognized archs, e.g., "SCMP_ARCH_X86_64"
+	// Nil value means "unknown", not "no support for any arch".
+	Archs []string `json:"archs,omitempty"`
+}
+
+// Apparmor represents the "apparmor" field
+type Apparmor struct {
+	// Enabled is true if AppArmor support is compiled in
+	// Unrelated to whether the host supports AppArmor or not.
+	// Always true in the current version of runc.
+	Enabled bool `json:"enabled"`
+}
+
+// Selinux represents the "selinux" field
+type Selinux struct {
+	// Enabled is true if SELinux support is compiled in
```

Comment:
ditto

Reply:
Does it make sense to add libselinux version here?

Reply:
No, because not linked in runc.

Reply:
You mean, statically linked? Yet it is linked, just dynamically, and runc will show whatever version is found by ld.so.

Not sure if it's needed though, just thinking out loud.

Reply:
Not linked at all, AFAICS

```console
$ ldd runc
        linux-vdso.so.1 (0x00007fffaf3e6000)
        libseccomp.so.2 => /lib/x86_64-linux-gnu/libseccomp.so.2 (0x00007f1cab638000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f1cab410000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f1cac0fe000)
```

Reply:
Ughm, I mixed selinux with seccomp 😵‍💫. Surely we either support selinux or not, and opencontainers/selinux (unlike seccomp/libseccomp-golang) does not use any external dependency.